test/runtests.jl

```diff
@@ -492,6 +492,186 @@ end
+        println("Verifying table existence for TPCH tables...")
+        for table in expected_tables
+            exists = RustyIceberg.table_exists(catalog, ["tpch.sf01"], table)
+            @test exists == true
+        end
+        println("✅ All TPCH tables verified to exist: $expected_tables")
+
+        # Test that a non-existent table returns false
+        nonexistent_exists = RustyIceberg.table_exists(catalog, ["tpch.sf01"], "nonexistent_table")
+        @test nonexistent_exists == false
+        println("✅ Non-existent table correctly returns false")
+
+        # Test listing tables in incremental namespace
+        println("Attempting to list tables in incremental...")
+        incremental_tables = RustyIceberg.list_tables(catalog, ["incremental"])
+        @test isa(incremental_tables, Vector{String})
+        println("✅ Tables in incremental: $incremental_tables")
+
+        # Test table existence check for incremental tables
+        println("Verifying table existence for incremental tables...")
+        for table in incremental_tables
+            exists = RustyIceberg.table_exists(catalog, ["incremental"], table)
+            @test exists == true
+        end
+        println("✅ All incremental tables verified to exist: $incremental_tables")
+
+    catch e
+        rethrow(e)
+    finally
+        # Clean up
+        if catalog != C_NULL
+            RustyIceberg.free_catalog(catalog)
+            println("✅ Catalog cleaned up successfully")
+        end
+    end
+
+    println("✅ Catalog API tests completed!")
+end
+
+@testset "Catalog Table Loading" begin
+    println("Testing catalog table loading...")
+
+    catalog_uri = "http://localhost:8181/api/catalog"
+
+    catalog = C_NULL
+    table = C_NULL
+    scan = C_NULL
+    stream = C_NULL
+    batch = nothing
+
+    try
+        # Create catalog connection with MinIO S3 configuration
+        # Note: Use localhost since we're running from the host, not from within the Docker network
+        props = Dict(
+            "credential" => "root:s3cr3t",
+            "scope" => "PRINCIPAL_ROLE:ALL",
+            "warehouse" => "warehouse",
+            "s3.endpoint" => "http://localhost:9000",
+            "s3.access-key-id" => "root",
+            "s3.secret-access-key" => "password",
+            "s3.region" => "us-east-1"
+        )
+        catalog = RustyIceberg.catalog_create_rest(catalog_uri; properties=props)
+        @test catalog != C_NULL
+        println("✅ Catalog created successfully")
+
+        # Load the customer table from tpch.sf01 namespace
+        println("Attempting to load customer table from tpch.sf01...")
+        table = RustyIceberg.load_table(catalog, ["tpch.sf01"], "customer")
+        @test table != C_NULL
+        println("✅ Customer table loaded successfully from catalog")
+
+        # Create a scan on the loaded table
+        println("Creating scan on loaded customer table...")
+        scan = RustyIceberg.new_scan(table)
```

Comment:
maybe here it'd be nice to have one test that does incremental scan on a table from catalog